tests/testthat/test-independent-testing-tlf_ae_specific.R

```diff
@@ -6,18 +6,304 @@ outdata <- prepare_ae_specific(meta,
+    extend_ae_specific_inference() |>
+    extend_ae_specific_events() |>
+    extend_ae_specific_duration(duration_var = "ADURN") |>
+    format_ae_specific(display = c("events", "dur", "n", "prop",
+      "diff", "diff_p", "diff_ci"), mock = TRUE) |>
+    tlf_ae_specific(
+      medra_version = "24.0",
+      source = "Source:  [CDISCpilot: adam-adsl; adae]",
+      path_outdata =  path_rdata,
+      path_outtable =  path_rtf
+    )
+
+  testthat::expect_snapshot_file(path_rtf)
+})
+
+test_that("rtf output: diff, events, dur, n, and prop w/ total", {
+  path_rtf <- file.path(tempdir(), "diednpt_ae0specific1.rtf")
+  path_rdata <- tempfile(fileext = '.Rdata')
+
+  tbl <- outdata |>
+    extend_ae_specific_inference() |>
+    extend_ae_specific_events() |>
+    extend_ae_specific_duration(duration_var = "ADURN") |>
+    format_ae_specific(display = c("events", "dur", "n", "prop", "total",
+      "diff", "diff_p", "diff_ci"), mock = TRUE) |>
+    tlf_ae_specific(
+      medra_version = "24.0",
+      source = "Source:  [CDISCpilot: adam-adsl; adae]",
+      path_outdata =  path_rdata,
+      path_outtable =  path_rtf
+    )
+
+  testthat::expect_snapshot_file(path_rtf)
+})
+
+#
+# test_that("rtf output: diff, n, and prop w/ total", {
+#   path_rtf <- file.path(tempdir(), "dinpt_ae0specific1.rtf")
+#   path_rdata <- tempfile(fileext = '.Rdata')
+#
+#   tbl <- outdata |>
+#     extend_ae_specific_inference() |>
+#     format_ae_specific(display = c("n", "prop", "total",
+#       "diff", "diff_p", "diff_ci")) |>
+#     tlf_ae_specific(
+#       medra_version = "24.0",
+#       source = "Source:  [CDISCpilot: adam-adsl; adae]",
+#       path_outdata =  path_rdata,
+#       path_outtable =  path_rtf
+#     )
+#
+#   testthat::expect_snapshot_file(path_rtf)
+# })
+#
+# test_that("rtf output: diff, n, and prop w/o total", {
+#   path_rtf <- file.path(tempdir(), "dinp_ae0specific1.rtf")
+#   path_rdata <- tempfile(fileext = '.Rdata')
+#
+#   tbl <- outdata |>
+#     extend_ae_specific_inference() |>
+#     format_ae_specific(display = c("n", "prop", "total",
+#       "diff", "diff_p", "diff_ci")) |>
+#     tlf_ae_specific(
+#       medra_version = "24.0",
+#       source = "Source:  [CDISCpilot: adam-adsl; adae]",
+#       path_outdata =  path_rdata,
+#       path_outtable =  path_rtf
+#     )
+#
+#   testthat::expect_snapshot_file(path_rtf)
+# })
+
+test_that("relative width 'works'", {
+  path_rtf <- file.path(tempdir(), "dinp_ae0specific1.rtf")
+  path_rdata <- tempfile(fileext = '.Rdata')
+
+  expect_error({
+  tbl <- outdata |>
+    format_ae_specific(display = c("n", "prop"), mock = TRUE) |>
+    tlf_ae_specific(
+      medra_version = "24.0",
+      source = "Source:  [CDISCpilot: adam-adsl; adae]",
+      path_outdata =  path_rdata,
+      path_outtable =  path_rtf,
+      col_rel_width = c(rep(1, 8))
+    )},
+    regexp = "col_rel_width must have the same length"
   )
 
-test_that("ae0specific1.rtf file output", {
-  expect_snapshot(tbl)
```

Comment:
previous testing was quite weak, and did not really track the created object in any way. Also didn't really explore most of the expected functionality of the `tlf_ae_specific()` function.
current setup allows us to view the rft snapshot and decide whether it looks correct, then to track that. 
swapped to `mock = TRUE` to lower the size of the snapshot files.